pkg/scheduling/v1/queuer.go

```diff
@@ -410,6 +497,7 @@ func (q *Queuer) ack(r *assignResults) {
 	for _, rateLimitedItem := range r.rateLimited {
```

Comment:
Since we are now making these statements dependent on ordering (i.e. we must process `r.rateLimited` last for this to work), can we add a warning here to indicate that the ordering shouldn't be changed?